### DISCUSSION TITLE
How to use parameters in attachments or blocks with the Slack trigger?

### DISCUSSION THREAD

**Discussion by ryok-0319** [argoproj/argo-events#2549] [Q&A]
## Background
I'm currently developing a sensor that sends messages to Slack using attachments or blocks
Depending on the payload of the event, I would like to dynamically adjust the content of the message that is sent

I'm aware parameters are supported in the Slack trigger. However, since attachments and blocks are JSON in YAML format, I haven't found a good way to use parameters in them

Is there a recommended or efficient approach to achieve this?

## Example
```
apiVersion: argoproj.io/v1alpha1
kind: Sensor
metadata:
  name: webhook
spec:
  dependencies:
    - name: test-dep
      eventSourceName: webhook
      eventName: example
  triggers:
    - parameters:
        - src:
            dependencyName: test-dep
            dataKey: body.channel
          dest: slack.channel
        - src:
            dependencyName: test-dep
            dataKey: body.message
          dest: slack.message
      template:
        name: slack-trigger
        slack:
          slackToken:
            key: token
            name: slack-secret
          channel: general
          attachments: |
            [{
              "title": "Attachment1!",
              "title_link": "https://argoproj.github.io/argo-events/sensors/triggers/slack-trigger/",
              "color": "#18be52",
              "fields": [{
                "title": "Hello1",
                "value": "Hello World1",
                "short": true
              }, {
                "title": "Hello2",
                "value": "Hello World2",
                "short": true
              }]
            }]
```

The manifest above is a copy of [one of the exmaples](https://github.com/argoproj/argo-events/blob/master/examples/sensors/slack-trigger.yaml) with a few changes

For example, if I want to change the title of the attachment depending on the payload of the event, is there a way to make this work?

**Comment by ryok-0319** [comment]
I solved this problem by creating a custom trigger
It works similarly to the official Slack trigger, but supports message templating
If a parameter exists that corresponds to a variable in the template, that variable will be expanded in the template

**Reply by guillaume-streitmatter-ext** [reply]
Hello @ryok-0319 Could give us an example of what you found as a solution? It would be very helpful.
Thank you :)
